corehq/apps/custom_data_fields/models.py

```diff
@@ -90,6 +93,11 @@ def validate_choices(field, value):
                     options=', '.join(field.choices),
                 )
 
+        def validate_regex(field, value):
+            if field.regex and value and not re.search(field.regex, value):
```

Comment:
Did you consider using `match` instead of `search`? In the example in the PR, input like `01234567890` or `a0123456789` will be allowed.

Reply:
I thought `re.search` had more intuitive behavior - neither validates the whole string unless you use `^` and `$`, and `re.match` starts at the beginning of the string, while `re.search` searches anywhere.
```python
In [1]: import re

In [2]: test_strings = ["123", "a123", "123a", "a123a", "abcde"]

In [3]: [bool(re.match("[1-9]{3}", test_string)) for test_string in test_strings]
Out[3]: [True, False, True, False, False]

In [4]: [bool(re.search("[1-9]{3}", test_string)) for test_string in test_strings]
Out[4]: [True, True, True, True, False]

In [5]: [bool(re.match("^[1-9]{3}$", test_string)) for test_string in test_strings]
Out[5]: [True, False, False, False, False]

In [6]: [bool(re.search("^[1-9]{3}$", test_string)) for test_string in test_strings]
Out[6]: [True, False, False, False, False]
```

Reply:
Huh, I had always thought re.match searched the whole string. Learned something new. That all makes sense to me, then - though if there are docs for this flag it might be a good thing to call out.